Automattic/wp-calypso
# Gutenberg: Add "Repeat Visitor" block

#### Changes proposed in this Pull Request

The `Repeat Visitor` block enables the user to control the visibility of its nested block(s) depending on how many times a user has visited the website. This is @Automattic/opportunity's first Gutenberg block, so any feedback would be much appreciated.

<img width="984" alt="screen-shot-2019-01-25-at-4 25 32-pm" src="https://user-images.githubusercontent.com/4044428/51945237-5b6a1680-23db-11e9-9803-c983113a87ed.png">

<strong>For copy and design reviewers</strong>: Please let us know how we could refine the presentation of this block to our users. ~(I personally don't think `Visited` should be the final name for this block, among other things.)~

You might like to reference this internal master thread: p9xfpQ-iN.

#### Testing instructions

1. Spin up a [Jurassic Ninja instance](https://jurassic.ninja/create/?gutenpack&calypsobranch=try/visited-block&branch=master).

2. Connect Jetpack to your WPCOM account.

3. Navigate to Settings->Jetpack Constants, check JETPACK_BETA_BLOCKS, and Save.

4. Add the Repeat Visitor block to a post or page. Nest whatever block within the Repeat Visitor block.

5. Save a post and open it. 

6. Check the cookie value for `jp-visit-counter`; it should have incremented from `0` to `1`. 

7. Refresh the page multiple times and ensure that the `jp-visit-counter` continues to increment. Ensure that your content becomes visible after it crosses the minimum view threshold (and vice versa for maximum view threshold).
